packages/react-reconciler/src/ReactFiberClassComponent.js

```diff
@@ -465,6 +465,19 @@ export default function(
     instance.refs = emptyObject;
     instance.context = getMaskedContext(workInProgress, unmaskedContext);
 
+    const noSetStatesFromProps = !(instance.state === instance.props);
+    warning(
+      noSetStatesFromProps,
```

Comment:
I know this is consistent with the rest of the warnings, but I wonder if we should just inline `noSetStatesFromProps` and avoid the whole negation thing?

```js
warning(
  instance.state !== instance.props,
  `...`
)
```

Reply:
That looks better to me, as well. Should I make the edit?

Reply:
the guidance is not clear here tbh. i've seen @gaearon also prefer `warning(false,....` and doing all the logic in an if block. ¯\_(ツ)_/¯

Reply:
Typically we do `warning(false,  "")` when we also need to execute some other code when issuing the warning, such as setting a flag so that the warning is deduped.